pkg/virt-handler/vm_test.go

```diff
@@ -2582,8 +2585,7 @@ var _ = Describe("VirtualMachineInstance", func() {
 				PodIP:  podIPs[0],
 				PodIPs: podIPs,
 			}
-			err = controller.networkCacheStoreFactory.CacheForVMI(vmi).Write(interfaceName, podCacheInterface)
-			Expect(err).ToNot(HaveOccurred())
+			controller.networkController.CachePodInterfaceVolatileData(vmi, interfaceName, podCacheInterface)
```

Comment:
The original test used the non-volatile cache. Do we have unit test coverage for it?

Reply:
The non-volatile was used to inject data to see if reports the right thing. I do not think the intention was to check this cache, but to check that the status update does its thing.

So replacing it with the volatile data is both faster and closer to the intention here.

Per what I saw, how the non-volatile cache is used is already covered by other tests, like the ones under https://github.com/kubevirt/kubevirt/blob/main/pkg/network/setup/network_test.go .

Looking at the reporting/status part, at the end of the refactoring some scenarios are checked against the volatile cache and some against the non-volatile one.
But I think this is a small part, as the non-volatile and volatile caches interactions is simple. Maybe it is worth adding a specific test that checks that the non-volatile cache is populated in the volatile cache on first usage.

Reply:
I have added a test to cover the last point [here](https://github.com/kubevirt/kubevirt/pull/6951/commits/f5bed9bd63a2f23ff56b070a33323f73e1055694).